source/slang/slang-reflection-api.cpp

```diff
@@ -1485,7 +1489,7 @@ namespace Slang
                 Index bindingRangeIndex = m_extendedInfo->m_bindingRanges.getCount();
                 SlangBindingType bindingType = SLANG_BINDING_TYPE_CONSTANT_BUFFER;
                 Index spaceOffset = -1;
-                bool usesIndirectAllocation = false;
+                bool shouldAllocDescriptorSet = true;
```

Comment:
The terminology here is a bit confusing (both in the original code and the new).

We probably need to be super clear about what is going on here, because there's really three distinct cases we are trying to juggle here:

1. The constant-buffer-like case. That includes `ConstantBuffer<X>` for D3D12/VK, but also includes `ParameterBlock<X>` for D3D11/GL. Any opaque types used by `X` "leak" into the resource usage of the container, but any uniform/ordinary data does not.

2. The descriptor-set-like case. This is obviously for `PB<X>` on D3D12/VK, and pretty much only those.

3. The CUDA/CPU "it's just memory" case. This covers both `PB<X>` and `CB<X>` for both CUDA and CPU, where they are basically just `X*`.

The old flag named `usesIndirectAllocation` was meant to detect/capture cases (2) (via `RegisterSpace`) and (3) (via `Uniform`).

The new flag named `shouldAllocDescriptorSet` seems to only capture case (2). AFAICT this seems to greatly change how binding ranges for CBs/PBs are reflected for CPU/CUDA, and I'm having trouble seeing how it could possibly work.

Reply:
I think I intend to run the `if (shouldAllocDescriptorSet)` case for case 3 here, because we need the step to reflect the CB<X> itself as a binding range. Inside the `if` block, we already have further sub `if` statements to not do anything for case 3, so this will not change the current behavior except that `CB<X>` will appear as a binding range in the its containing descriptor set.